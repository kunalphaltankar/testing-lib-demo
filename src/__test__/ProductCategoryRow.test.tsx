import { render, screen } from "@testing-library/react"
import ProductCategoryRow from "../components/ProductCategoryRow";
import PRODUCTS from "../models/stub/PRODUCTS";

describe("ProductCategoryRow", () => {
  it('should render as table row header', ()=> {
    const { container } = render(
      <table>
        <tbody>
          <ProductCategoryRow category={PRODUCTS[0].category} />
        </tbody>
      </table>,
    );
    expect(container).toMatchSnapshot();
  })

  it('should take two colums', ()=> {
     render(
      <table>
        <tbody>
          <ProductCategoryRow category={PRODUCTS[0].category} />
        </tbody>
      </table>,
    );

    const th = screen.getByText(PRODUCTS[0].category);
    expect(th).toHaveAttribute('colSpan', '2')
  })
})