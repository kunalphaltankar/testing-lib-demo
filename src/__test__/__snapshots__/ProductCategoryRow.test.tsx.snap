// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCategoryRow should render as table row header 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <th
          colspan="2"
        >
          Sporting Goods
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;
